src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -60,7 +61,14 @@ class BaseOptionsSelector extends Component {
                     return;
                 }
 
-                this.selectRow(focusedOption);
+                // We are handeling multiple Enter events to address the issue in https://github.com/Expensify/App/issues/18265
+                if (!isEnterDisabled) {
+                    isEnterDisabled = true;
+                    this.selectRow(focusedOption);
```

Comment:
We could also handle promise resolution similar to here - https://github.com/Expensify/App/blob/18c8f9a3a32661ee5d1b5ccdf2f9a57c9a4f4658/src/components/OptionRow.js#L158-L161

Since this.selectRow already returns a promise. It wouldn't be good to reset the block variable before the promise is resolved.